src/components/Sidebar/Sidebar.jsx

```diff
@@ -0,0 +1,107 @@
+/* eslint-disable react-hooks/exhaustive-deps */
```

Comment:
This is not recommended, please review your hooks

Reply:
This is always giving issues in any project, it cause infinite loops because force to include the state that you are changing on the useEffect array, so when you update the state the useEffect is triggered again and again.